src/platform/plugins/shared/discover/public/application/main/components/chart/chart_portals_renderer.tsx

```diff
@@ -271,21 +271,20 @@ const CustomChartSectionWrapper = ({
     !!layoutProps.chart && !layoutProps.chart.hidden
   );
 
-  const hasValidSession = !!unifiedHistogramProps.searchSessionId || isEsqlMode;
-  const isComponentVisible =
-    !!chartSectionConfig.Component && !!layoutProps.chart && !layoutProps.chart.hidden;
-
-  if (!hasValidSession) {
+  if (!api || !fetchParams || !hasValidFetchParams) {
```

Comment:
```suggestion
  if (!fetchParams || !hasValidFetchParams) {
```
Nit: similar suggestion.